src/vm/node_modules/VM.js

```diff
@@ -11009,7 +11026,11 @@ exports.reprovision = function (uuid, payload, options, callback)
             });
         }, function (cb) {
             // unmount <zonepath>/cores so dataset is not busy
-            zfs(['umount', vmobj.zonepath + '/cores'], log,
+            // unmounting using the mountpoint fails often with:
+            //      "cannot unmount '/zones/<vm_uuid>/cores': not a mountpoint"
+            // so use the dataset name instead of the mountpoint which seems to
+            // solve the issue
+            zfs(['unmount', vmobj.zpool + '/cores/' + uuid], log,
```

Comment:
The comment here is just not true. There have been literally hundreds of thousands of reprovisions where the alleged error does not occur. Stating that it happens "often" is at best, an overestimation.

The only reason the `not a mountpoint` error occurs is if the dataset is not mounted. And if the dataset is not mounted trying to unmount by the dataset name will return the error `not currently mounted`. So you've only exchanged one error message for another here.

Functionally, there is exactly zero difference here.